src/Dto/SpecialSponsorshipRequest.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Dto;
+
+use App\Entity\SpecialSponsorship;
+use Symfony\Component\Validator\Constraints as Assert;
+
+final class SpecialSponsorshipRequest
+{
+    /**
+     * @Assert\NotBlank()
+     * @Assert\Type(type="string")
+     */
+    public $label;
+
+    /**
+     * @Assert\NotBlank()
+     * @Assert\Type(type="float")
+     */
+    public $price;
+
+    /**
+     * @Assert\NotBlank()
+     */
+    public $description;
+
+    public function __construct(string $label = null, float $price = null, string $description = null)
+    {
+        $this->label = $label;
+        $this->price = $price;
+        $this->description = $description;
+    }
+
+    public static function createFromEntity(SpecialSponsorship $specialSponsorship): SpecialSponsorshipRequest
+    {
+        return new SpecialSponsorshipRequest(
+            $specialSponsorship->getLabel(),
+            $specialSponsorship->getPrice(),
+            $specialSponsorship->getDescription()
+        );
+    }
+    public function updateEntity(SpecialSponsorship $specialSponsorship): void
```

Comment:
You do assume the mutability of the object (then passed in by reference I would assume), and therefore your function is not a pure function (mathematically speaking). You can probably refactor that bit to remove state from here.

Reply:
I don't understand what is wrong here and what you want, can you be more specific ?